ironfish/src/primitives/transaction.ts

```diff
@@ -94,17 +94,17 @@ export class Transaction {
       // authorizing signature
       reader.seek(64)
 
-      return new MintDescription(asset, value)
+      return { asset, value }
     })
 
     this._burns = Array.from({ length: _burnsLength }, () => {
       const asset = Asset.deserialize(reader.readBytes(ASSET_LENGTH))
-      const value = reader.readU8()
+      const value = BigInt(reader.readU64())
```

Comment:
This is a bug, readU64() cannot read greater than 2^53. You need to use the big versions of these functions.

Reply:
Yeah, I think that makes sense. Would [this](https://github.com/iron-fish/ironfish/blob/master/ironfish/src/primitives/note.ts#L25) be a bug as well then?

Reply:
Yep, there are bigint versions of all these functions in bufio. If we have a number, and were casting to BigInt or from BigInt we should almost always treat that as an error unless you can guarantee it won't overflow like the miners fee.